uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/PublicHolidayLeaveRequestCreationTest.php

```diff
@@ -380,6 +421,10 @@ public function testItCreatesLeaveRequestsForAllContactsWithContractsOverlapping
     $contact1 = ContactFabricator::fabricate();
     $contact2 = ContactFabricator::fabricate();
 
+    AbsenceType::del($this->absenceType->id);
+    $absenceType1 = AbsenceTypeFabricator::fabricate(['must_take_public_holiday_as_leave' => 1]);
+    $absenceType2 = AbsenceTypeFabricator::fabricate(['must_take_public_holiday_as_leave' => 1]);
+
```

Comment:
This also seems to be repeated in multiple tests. Please check if it's possible to avoid that

Reply:
Not sure what you mean and how we can avoid this. I will add a comment for why existing absence types are deleted like you mentioned, but for most cases in the tests, the absence types ids need to be used in the actual tests and would not make the test clear if these were extracted to a function.